Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/ChannelGroup/ByChannelGroup.tsx

```diff
@@ -0,0 +1,167 @@
+//  ----------------------------------------------------------------------------------------------------
+//  06/30/2023 - Parker Dinsdale
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+
+import * as React from 'react';
+import { useAppSelector, useAppDispatch } from '../hooks';
+import { ChannelGroupSlice, ChannelGroupDetailsSlice } from '../Store/Store';
+
+import Table from '@gpa-gemstone/react-table'
+import * as _ from 'lodash';
+import { useHistory } from "react-router-dom";
+import { SystemCenter, Application } from '@gpa-gemstone/application-typings';
+import { SearchBar, Search, Modal } from '@gpa-gemstone/react-interactive';
+
+import { DefaultSearchField, SearchFields, TransformSearchFields } from '../CommonComponents/SearchFields';
+import ChannelGroupForm from './ChannelGroupForm';
+
+
+const ChannelGroups: Application.Types.iByComponent = (props) => {
+    const dispatch = useAppDispatch();
+
+    const data = useAppSelector(ChannelGroupSlice.SearchResults);
+    const status = useAppSelector(ChannelGroupSlice.SearchStatus);
+    const items = useAppSelector(ChannelGroupDetailsSlice.Data);
+    const itemStatus = useAppSelector(ChannelGroupDetailsSlice.Status);
+    const parentID = useAppSelector(ChannelGroupDetailsSlice.ParentID);
+
+    const [showNew, setShowNew] = React.useState<boolean>(false);
+    const [sortField, setSortField] = React.useState<keyof SystemCenter.Types.ChannelGroup>('Name');
+    const [ascending, setAscending] = React.useState<boolean>(true);
+
+    const emptyRecord = { ID: 0, Name: '', Description: '' };
+    let history = useHistory();
+
+    const [search, setSearch] = React.useState<Array<Search.IFilter<SystemCenter.Types.ChannelGroup>>>([]);
+
+    const [record, setRecord] = React.useState<SystemCenter.Types.ChannelGroup>(emptyRecord);
+
+    React.useEffect(() => {
+        if (status == 'unintiated' || status == 'changed')
+            dispatch(ChannelGroupSlice.DBSearch({ filter: search, sortField, ascending }));
+    }, [dispatch, status]);
+
+    React.useEffect(() => {
+        dispatch(ChannelGroupSlice.DBSearch({ filter: search, sortField, ascending }));
+    }, [search, sortField, ascending]);
+
+    React.useEffect(() => {
+        if (itemStatus == 'unintiated' || itemStatus == 'changed' || parentID != null)
+            dispatch(ChannelGroupDetailsSlice.Fetch());
+    }, [dispatch, itemStatus, parentID]);
+
+    function handleSelect(item) {
+        history.push({ pathname: homePath + 'index.cshtml', search: '?name=ChannelGroup&GroupID=' + item.row.ID })
+    }
+
+    return (
+        <div style={{ width: '100%', height: '100%' }}>
+            <SearchBar< SystemCenter.Types.ChannelGroup>
+                CollumnList={SearchFields.ValueListGroup as Search.IField<SystemCenter.Types.ChannelGroup>[]}
+                SetFilter={(flds) => setSearch(flds)}
```

Comment:
```suggestion
                SetFilter={(flds) => dispatch(ChannelGroupSlice.DBSearch({ filter: flds }));}
```
then you  can switch `search` to come from the Slice